debezium-core/src/main/java/io/debezium/relational/TableIdParser.java

```diff
@@ -0,0 +1,231 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.relational;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import io.debezium.text.ParsingException;
+import io.debezium.text.TokenStream;
+import io.debezium.text.TokenStream.CharacterStream;
+import io.debezium.text.TokenStream.Tokenizer;
+import io.debezium.text.TokenStream.Tokens;
+
+/**
+ * Parses identifiers into the corresponding parts of a {@link TableId}.
+ *
+ * @author Gunnar Morling.
+ */
+class TableIdParser {
+
+    private static final char SEPARATOR = '.';
+
+    public static List<String> parse(String identifier, char quotingChar) {
+        TokenStream stream = new TokenStream(identifier, new TableIdTokenizer(identifier, quotingChar), true);
+        stream.start();
+
+        List<String> parts = new ArrayList<>();
+
+        while(stream.hasNext()) {
+            parts.add(stream.consume());
+        }
+
+        return parts;
+    }
+
+    private static class TableIdTokenizer implements Tokenizer {
```

Comment:
@rhauch thanks a lot, you've pushed me into the right direction. I got rid of the notion of `IdentifierTransformer` - when table ids are parsed, the quoting char is derived automatically now. Only when quoting a table name, it must naturally be given as it's DB-dependent, so `TableId#toQuotedString()` expects the quoting char as an input.

In terms of parsing, I agree we should probably use `BasicTokenizer` eventually. For the time being I'm still using my own implementation, though (which I've adapted to return unquoted name parts, as done by `BasicTokenizer`). The reason being that `BasicTokenizer removes escaped quoting characters: e.g. parsing `"tab""le"` will result in `table`, whereas it should be `tab"le` (so distinct topic names for `"tab""le"` and `"tabl""e"` can be derived).

Eventually I think this should be fixed within `BasicTokenizer` itself, but I'm running a bit out of time on this one so I decided to keep the other implementation for the time being. I'll create a JIRA issue for fixing the matter in the basic tokenizer and converge to one implementation later on.